app/tasks_routes.py

```diff
@@ -0,0 +1,147 @@
+from app import db
+from app.models.task import Task
+from flask import Blueprint, current_app, jsonify, abort, make_response, request
+from sqlalchemy import desc
+import requests
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix = "/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+def send_slack_message_for_completed_task(task_title):    
+    slack_api_token = current_app.config["SLACK_API_TOKEN"]
+    # slack_api_url = current_app.config["SLACK_API_URL"]
+    # slack_channel = current_app.config["SLACK_CHANNEL"]
+    slack_api_url = "https://slack.com/api/chat.postMessage"
+    slack_channel = "task-notifications"
```

Comment:
In Learn, I see that you added these explicit values because of issues getting the tests to run at submission. Without the `.env` file, these would end up being `None`, which `requests.post` may not have liked for the URL. While that's not necessarily a bad thing (a crash would tell us that we forgot a configuration value when deploying), we do need a workaround for the submission tests.

One way to accomplish this could be to read the value from the config, but if its falsy, fall back to a hard-coded default.

```py
    slack_api_url = current_app.config["SLACK_API_URL"] or "https://slack.com/api/chat.postMessage"
```

If the config value is truthy (as a non-empty string would be), then it will be used. If it's falsy, then the value on the right of `or` will be used.